File: src/13.ts
import { promises as fs } from 'fs'
import { range } from './share/iterable-helper'

const solve11 = async (): Promise<void> => {
    const rows1 = (await fs.readFile('./res/col1.txt', 'utf-8'))
        .split('\n')
    const rows2 = (await fs.readFile('./res/col2.txt', 'utf-8'))
        .split('\n')
    const merged =
        range(rows1.length)
            .map(i => `${rows1[i]}\t${rows2[i]}`)
            .join('\n')
    await fs.writeFile('./res/merged.txt', merged)
}

solve11()

File: src/13.py
# paste res/col1.txt res/col2.txt

from helper.file_helper import read_string, write_string

source = read_string('./res/popular-names.txt')
rows1 = read_string('./res/col1.txt')
rows2 = read_string('./res/col2.txt')
merged = '\n'.join(
    f'{row1}\t{row2}'
    for row1, row2 in zip(rows1.split('\n'), rows2.split('\n'))
)
write_string('./res/merged.txt', merged)
